gstreamer/src/tags.rs

```diff
@@ -554,6 +570,179 @@ impl StaticType for TagListRef {
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter.write_str("a tuple of 2 elements (name: String, value: Tag value type)")
+    }
+
+    fn visit_seq<A: SeqAccess<'de>>(self, mut seq: A) -> Result<Self::Value, A::Error> {
+        let name = CString::new(
+                seq
+                    .next_element::<String>()
+                    .map_err(|_|
+                        de::Error::custom("Expected a `String` name for Tag")
+                    )?
+                    .ok_or(de::Error::custom("Expected a value for Tag name, found `None`"))?
+                    .as_str()
+            ).unwrap();
+
+        unsafe {
+            let type_: glib::Type = from_glib(ffi::gst_tag_get_type(name.as_ptr() as *const i8));
+            let mut value = Value::from_type(type_);
+            match type_ {
+                glib::Type::String => {
+                    // TODO: use a macro
+                    glib::value::SetValue::set_value(
+                        &mut value,
+                        &seq
+                            .next_element::<String>()
+                            .map_err(|_|
+                                de::Error::custom(
+                                    format!("Expected value with type `{}` for Tag {:?}", type_, name),
+                                )
+                            )?
+                            .ok_or_else(||
+                                de::Error::custom(
+                                    format!("Expected a value for Tag {:?}, found `None`", name),
+                                )
+                            )?,
+                    );
+                }
+                _ => {
+                    return Err(
+                        de::Error::custom(
+                            format!("unimplemented deserialization for Tag {:?} with type `{}`",
+                                name,
+                                type_,
+                            ),
+                        )
+                    );
+                }
+            }
+
+            Ok(TagDe(name, value))
+        }
+    }
+}
+
+#[cfg(feature = "serde-support")]
+impl<'de> Deserialize<'de> for TagDe {
+    fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
+        deserializer.deserialize_tuple(2, TagVisitor{}) // 2 items in the tuple: (name, value)
+    }
+}
+
+#[cfg(feature = "serde-support")]
+struct TagListVisitor;
+
+#[cfg(feature = "serde-support")]
+impl<'de> Visitor<'de> for TagListVisitor {
+    type Value = TagList;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter.write_str("a sequence of Tags")
+    }
+
+    fn visit_seq<A: SeqAccess<'de>>(self, mut seq: A) -> Result<Self::Value, A::Error> {
+        let tag_list = TagList::new();
+        while let Some(tag_de) = seq.next_element::<TagDe>()? {
+            unsafe {
+                ffi::gst_tag_list_add_value(
```

Comment:
There should probably be some checking if the value is of the correct type for the tag here